akka-actor/src/main/scala/akka/io/dns/internal/AsyncDnsResolver.scala

```diff
@@ -93,17 +92,28 @@ private[io] final class AsyncDnsResolver(
           log.debug("{} cached {}", mode, resolved)
           sender() ! resolved
         case None =>
-          resolveWithResolvers(name, mode, resolvers)
-            .map { resolved =>
-              if (resolved.records.nonEmpty) {
-                val minTtl = (positiveCachePolicy +: resolved.records.map(_.ttl)).min
-                cache.put((name, mode), resolved, minTtl)
-              } else if (negativeCachePolicy != Never) cache.put((name, mode), resolved, negativeCachePolicy)
-              log.debug(s"{} resolved {}", mode, resolved)
-              resolved
+          val replyTo = sender()
+          resolveWithResolvers(name, mode, resolvers).onComplete {
```

Comment:
Broad idea is to limit the future callbacks inside the actor.  Combining the cache updates with piping preserves the sequencing in `.map { x => sideEffect(); x }.pipeTo(sender())` and allows use of `parasitic` for performance without making that the default context.

Reply:
maybe  pipeTo is better .

Reply:
Yes, +100 for `parasitic`